sdk/storage/Azure.Storage.Blobs/src/Generated/BlobRestClient.cs

```diff
@@ -43,7 +45,9 @@ public static partial class Service
                     timeout,
                     requestId))
                 {
-                    Azure.Response _response = await pipeline.SendRequestAsync(_request, cancellationToken).ConfigureAwait(false);
+                    Azure.Response _response = async ?
```

Comment:
Interesting approach.  We could use a discussion about how this avoids deadlocks (assuming it does -- we are still in an async Task-returning method).

Reply:
You can do this safely if the task has completed before you wait.  We've got a number of places in `Azure.Core` where we share code between sync/async methods this way.

Unfortunately our lack of a fully sync transport layer means we're still doing sync-over-async in the meantime.

Reply:
Could you add this information as comments here?  Or an assertion to verify that the task has completed?

Reply:
I don't think we need a comment here because nobody's going to read the generated REST code.

There's an assertion and comments in `TaskExtensions.EnsureCompleted` which we use on all our sync APIs.  (And test coverage has verified that we don't have any tasks not completing for our sync APIs across the board.)

Reply:
On the contrary, the generated REST code has been read a lot by us in debugging issues, and async vs sync issues are a high-profile and somewhat arcane set of problems.  I'd like a lot of hand-holding for devs on new patterns, especially external devs.

This is minor, but I think we should probably add them.  Given this *is* generated code, the change actually would be made elsewhere.

Reply:
discussed offline